osbs/api.py

```diff
@@ -166,7 +185,19 @@ def _create_build_config_and_build(self, build_json, namespace):
             # if it doesn't exist, then create it
             logger.debug('build config for %s doesn\'t exist, creating...', build_config_name)
             self.os.create_build_config(json.dumps(build_json), namespace=namespace)
-        return self.os.start_build(build_config_name, namespace=namespace)
+            # under some circumstances, creating a buildConfig might instantiate it, so
+            #  check for an already running build and return it if it exists
+            #  see https://github.com/projectatomic/osbs-client/issues/205
```

Comment:
My intention was to make this general, to have the code prepared for possible future changes in Openshift, that would change that. IMO it doesn't hurt to have the change general in this way.

My understanding was that creating buildconfig and instantiating build from it was an atomic operation and thus we could afford to do this.

But you're right. If this is not an atomic operation, then we need to know the precise circumstances when the build will be instantiated and wait for it.